src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -83,7 +83,14 @@ public virtual void ApplyRule(RewriteContext context)
                             throw new ArgumentOutOfRangeException($"Unexpected server variable: `{serverVariable.Type}`");
                     }
 
-                    headerDictionary.Append(name, value);
+                    if (headerDictionary.ContainsKey(name))
```

Comment:
i had to pull in the source to see what was happening. `.Append` is fine as you pointed out. the reason `.Append` fails my tests is because of what i noted in the test:

```csharp
// REMOTE_ADDR is not set by the test framework, so we assert the existence of the header rather than its value
```

which causes this to run:

https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/Internal/ParsingHelpers.cs#L104

so i'll change the tests to pass and restore `.Append`.

is there a way to get `TestServer` to actually hydrate `HttpContext.Connection.RemoteIpAddress`?